libraries/bezier-rs/src/subpath/transform.rs

```diff
@@ -359,25 +367,42 @@ impl<ManipulatorGroupId: crate::Identifier> Subpath<ManipulatorGroupId> {
 			let angle = out_tangent.angle_between(in_tangent);
 
 			// The angle is concave. The Subpath overlap and must be clipped
-			let mut apply_joint = true;
+			let mut apply_join = true;
 			if (angle > 0. && distance > 0.) || (angle < 0. && distance < 0.) {
 				// If the distance is large enough, there may still be no intersections. Also, if the angle is close enough to zero,
 				// subpath intersections may find no intersections. In this case, the points are likely close enough that we can approximate
 				// the points as being on top of one another.
 				if let Some((clipped_subpath1, clipped_subpath2)) = Subpath::clip_simple_subpaths(subpath1, subpath2) {
 					subpaths[i] = clipped_subpath1;
 					subpaths[j] = clipped_subpath2;
-					apply_joint = false;
+					apply_join = false;
 				}
 			}
-			// The angle is convex. The Subpath must be joined using the specified Joint type
-			if apply_joint {
-				match joint {
-					Joint::Bevel => {
+			// The angle is convex. The Subpath must be joined using the specified join type
+			if apply_join {
+				match join {
+					Join::Bevel => {
+						drop_common_point[j] = false;
+					}
+					Join::Miter => {
+						let miter_manipulator_group = subpaths[i].miter_line_join(&subpaths[j]);
+						if let Some(miter_manipulator_group) = miter_manipulator_group {
+							subpaths[i].manipulator_groups.push(miter_manipulator_group);
+						}
+						drop_common_point[j] = false;
+					}
+					Join::Round => {
+						let (out_handle, round_point, in_handle) = subpaths[i].round_line_join(&subpaths[j], self.manipulator_groups[j].anchor);
+						let last_index = subpaths[i].manipulator_groups.len() - 1;
+						subpaths[i].manipulator_groups[last_index].out_handle = Some(out_handle);
+						subpaths[i].manipulator_groups.push(round_point.clone());
+						subpaths[j].manipulator_groups[0].in_handle = Some(in_handle);
 						drop_common_point[j] = false;
 					}
-					_ => unimplemented!(),
 				}
+			} else {
+				// Otherwise, default to the bevel join
+				drop_common_point[j] = false;
```

Comment:
We seem to be setting `drop_common_point[j] = false` in every arm of the if-statement and match-block. Do we even need this boolean array anymore? The original goal of this boolean array was to drop duplicate points in cases where the outline lined up perfectly (ie. no joining / clipping necessary). Could you check to make sure we're still doing that?

Reply:
The condition on line 360 should still be catching that right?

Reply:
Can you extract the `drop_common_point[j] = false;` to the top of the if-statement, since you're setting it true in every arm of the match statement anyway?